src/engraving/dom/textbase.cpp

```diff
@@ -1000,8 +1000,10 @@ void TextBlock::layout(const TextBase* t)
         }
         break;
         case ElementType::MEASURE: {
+            // ignore courtesy keysig, timesig, but fall back if needed
             Measure* m = toMeasure(e);
-            layoutWidth = m->ldata()->bbox().width();
+            Segment* s = m->findSegment(SegmentType::EndBarLine, m->tick() + m->ticks());
+            layoutWidth = s ? s->x() : m->width();
```

Comment:
why dropping the `tdata()->bbox()` here?

Reply:
using the `width` method gave me identical results while testing, but maybe I missed something...

Reply:
May not be needed, but doesn't look like a full fall back as announced in the comment

Reply:
Try `layoutWidth = s ? s->x() + s->width() : m->width();` to fix the vtests

Reply:
Already found that fix, for now waiting to see if the changes are desired or not. but thanks :)

Reply:
Hmm, seems to go a bit too far right with that in the case you're trying to fix:
![image](https://github.com/musescore/MuseScore/assets/1786669/050d52c2-7417-40c0-9ef4-53db6924a457)
But still better (i.e. same as before) when there's no courtesy key- or time sig or clef.

Reply:
Looks line the barline to key-sig distance being added?

Reply:
Yes, its included in the segment width so we'd have to use only the barline width here